### PR TITLE
Create RuntimeCommand and remove phel-composer-plugin dependency

### PR DESCRIPTION
**Pull request by Chemaclass**
## 💡 Goal

Move all phel config to `phel-config.php` and deprecate the composer extra config fields in favor of the PHP phel config.

## 📚 Description

We want to get rid of the `phel-composer-plugin` dependency and move all project config to `phel-config.php`. And follow up we could start thinking about how could we define the config in a phel itself. Kind of `phel-config.phel`. But this is out of the scope of this PR.

## 🔖 Changes

- Create `RuntimeCommand`
- Remove `phel-composer-plugin` dependency from `composer.json`

### CODE REVIEW

File: src/php/Command/Runtime/ConfigNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Command\Runtime;
+
+final class ConfigNormalizer
+{
+    /**
+     * @param array<string, list<string>> $result [ns => [path1, path2, ...]]
```
Comment (jenshaase) [inline src/php/Command/Runtime/ConfigNormalizer.php:10]: Could you describe what this function is doing?

What type of `array` is `$configLoader`?

File: src/php/Command/Runtime/RuntimeCommand.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        return $result;
+    }
+
+    /**
+     * @return array<string, list<string>>
+     */
+    private function loadVendorConfig(array $result): array
+    {
+        $pattern = $this->applicationRootDir . 'vendor/*/*/' . self::PHEL_CONFIG_FILE_NAME;
```
Comment (jenshaase) [inline src/php/Command/Runtime/RuntimeCommand.php:78]: In composer it is possible to change the location of the `vendor` directory (https://getcomposer.org/doc/06-config.md#vendor-dir). Therefore, we should have an options to change the `vendor` directory too. I would suggest we have a config value for that. Default to `vendor`'

File: src/php/Command/Runtime/RuntimeCommand.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * @return array<string, list<string>>
+     */
+    private function loadRootConfig(array $result): array
+    {
+        $rootPhelConfigPath = $this->applicationRootDir . self::PHEL_CONFIG_FILE_NAME;
+
+        if (is_file($rootPhelConfigPath)) {
+            $pathPrefix = '/..';
```
Comment (jenshaase) [inline src/php/Command/Runtime/RuntimeCommand.php:64]: Can you explain this to me. Why this value?

**Review by jenshaase — CHANGES_REQUESTED** [review]

File: src/php/Command/Runtime/RuntimeCommand.php (hunk 3 above)
Comment (Chemaclass) [reply src/php/Command/Runtime/RuntimeCommand.php:64]: I took that value from here: https://github.com/phel-lang/phel-composer-plugin/blob/master/src/DumpRuntime.php#L63 when `isRootPackage === true`

```php
$pathPrefix = $isRootPackage ? '/..' : '/' . $package->getName();
```

File: src/php/Command/Runtime/ConfigNormalizer.php (hunk 1 above)
Comment (Chemaclass) [reply src/php/Command/Runtime/ConfigNormalizer.php:10]: I will do that 👍

File: src/php/Command/Runtime/RuntimeCommand.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        $loaderConfig = [
+            $this->loadRootConfig(),
+            $this->loadVendorConfig(),
+        ];
+
+        /** @var array<string, list<string>> $flatConfig [ns => [path1, path2, ...]] */
+        $flatConfig = array_merge(...$loaderConfig);
+
+        file_put_contents(
+            $this->applicationRootDir . 'vendor/PhelRuntime.php',
```
Comment (jenshaase) [inline src/php/Command/Runtime/RuntimeCommand.php:60]: We need to replace `vendor` here too.

**Review by jenshaase — CHANGES_REQUESTED** [review]

File: src/php/Command/Runtime/RuntimeCommand.php (hunk 4 above)
Comment (Chemaclass) [reply src/php/Command/Runtime/RuntimeCommand.php:60]: Right, thanks! I will change it

**Review by jenshaase — APPROVED** [review]

**Pull request merged**